AutoGear.lua

```diff
@@ -1563,7 +1563,11 @@ end
 
 function AutoGearSetStatWeights()
 	local localizedClass, class, spec = AutoGearGetClassAndSpec()
-	AutoGearCurrentWeighting = AutoGearDefaultWeights[class][spec] or nil
+	if AutoGearDefaultWeights[class] then
+		AutoGearCurrentWeighting = AutoGearDefaultWeights[class][spec]
```

Comment:
I think this line needs ` or nil` appended for the case where `AutoGearDefaultWeights[class]` is not `nil` but `AutoGearDefaultWeights[class][spec]` is `nil`.

By the way, what does this change accomplish?  I thought the original line would result in `nil` even in the case where `AutoGearDefaultWeights[class]` is `nil`.  Was I incorrect there?

Also, could this whole new block be written in a one-liner like this instead, like the `weapons` assignment below it?

```Lua
AutoGearCurrentWeighting = AutoGearDefaultWeights[class] and (AutoGearDefaultWeights[class][spec] or nil) or nil
```